library/core/src/const_closure.rs

```diff
@@ -39,25 +42,37 @@ impl<'a, CapturedData: ?Sized, Function> ConstFnMutClosure<'a, CapturedData, Fun
     }
 }
 
-impl<'a, CapturedData: ?Sized, ClosureArguments, Function, ClosureReturnValue> const
-    FnOnce<ClosureArguments> for ConstFnMutClosure<'a, CapturedData, Function>
-where
-    Function:
-        ~const Fn(&mut CapturedData, ClosureArguments) -> ClosureReturnValue + ~const Destruct,
-{
-    type Output = ClosureReturnValue;
+macro_rules! impl_fn_mut_tuple {
+    ($($var:ident)*) => {
+      #[allow(unused_parens)]
+      impl<'a, $($var,)* ClosureArguments, Function, ClosureReturnValue> const
+        FnOnce<ClosureArguments> for ConstFnMutClosure<($(&'a mut $var),*), Function>
+      where
+        Function: ~const Fn(($(&mut $var),*), ClosureArguments) -> ClosureReturnValue+ ~const Destruct,
+      {
+        type Output = ClosureReturnValue;
 
-    extern "rust-call" fn call_once(mut self, args: ClosureArguments) -> Self::Output {
-        self.call_mut(args)
-    }
-}
+        extern "rust-call" fn call_once(mut self, args: ClosureArguments) -> Self::Output {
+          self.call_mut(args)
+        }
+      }
+      #[allow(unused_parens)]
+      impl<'a, $($var,)* ClosureArguments, Function, ClosureReturnValue> const
+        FnMut<ClosureArguments> for ConstFnMutClosure<($(&'a mut $var),*), Function>
+      where
+        Function: ~const Fn(($(&mut $var),*), ClosureArguments)-> ClosureReturnValue,
+      {
+        extern "rust-call" fn call_mut(&mut self, args: ClosureArguments) -> Self::Output {
+          #[allow(non_snake_case)]
+          let ($($var),*) = &mut self.data;
+          (self.func)(($($var),*), args)
+        }
+      }
 
-impl<'a, CapturedData: ?Sized, ClosureArguments, Function, ClosureReturnValue> const
-    FnMut<ClosureArguments> for ConstFnMutClosure<'a, CapturedData, Function>
-where
-    Function: ~const Fn(&mut CapturedData, ClosureArguments) -> ClosureReturnValue,
-{
-    extern "rust-call" fn call_mut(&mut self, args: ClosureArguments) -> Self::Output {
-        (self.func)(self.data, args)
-    }
-}
+    };
+  }
```

Comment:
Sorry for the nits, but this isn't formatted properly. The last brace should have no spaces before them, and all of these should have four spaces.

r=me once you are done